sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -404,12 +417,12 @@ export class JsonRpcProvider extends Provider {
     }
   }
 
-  async getObjectRef(objectId: string): Promise<SuiObjectRef | undefined> {
+  async getObjectRef(objectId: ObjectId): Promise<SuiObjectRef | undefined> {
     const resp = await this.getObject(objectId);
     return getObjectReference(resp);
   }
 
-  async getObjectBatch(objectIds: string[]): Promise<GetObjectDataResponse[]> {
+  async getObjectBatch(objectIds: ObjectId[]): Promise<GetObjectDataResponse[]> {
     const requests = objectIds.map((id) => ({
```

Comment:
We should also validate each of the `objectIds` here. If one of them is invalid, feel free to throw an error that tells which of the `objectIds` is invalid.